mm2src/mm2_bitcoin/keys/src/address.rs

```diff
@@ -97,13 +99,15 @@ pub fn detect_checksum(data: &[u8], checksum: &[u8]) -> Result<ChecksumType, Err
 /// Struct for utxo address types representation
 /// Contains address hash, format, prefix to get as a string.
 /// Also has output ScriptType field to create output script.
-#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+#[derive(Clone, Debug, Eq)]
 pub struct Address {
     /// The base58 prefix of the address.
     prefix: AddressPrefix,
     /// Segwit addr human readable part
     hrp: Option<String>,
-    /// Public key hash.
+    /// The public key of the address.
+    pubkey: Option<Public>,
```

Comment:
Do we need to store pubkey in the Address? I believe an address is always hash of a script. For p2pk and p2pkh output their address is the same.
(I can see further in the code you only convert this pubkey to the hash)

Reply:
Yup we have to store it. We can't fetch p2pk balance from electurm without it [see](https://github.com/mariocynicys/adex/blob/f11e0ed65b1bdef4f18c6600440569b88adfb4d6/mm2src/coins/utxo/rpc_clients.rs#L2224-L2228).

Note that listing p2pk balance from `Native` client isn't implemented here (as discussed with @shamardy in DMs). Native is adjusted tho to report the output script of the utxo and have it stored in `UtxoInfo` so that we can check against it before signing it for spending.

Reply:
this is how script pubkey for p2pk output looks like (`21` + `pubkey` + `ac`):
![image](https://github.com/KomodoPlatform/komodo-defi-framework/assets/56920956/c621a818-9eb7-4f00-9781-1752e544e3c8)

which is different from p2pkh ones (`76a914` + `pubkey_hash` + `88ac`):
![image](https://github.com/KomodoPlatform/komodo-defi-framework/assets/56920956/1b37efbe-64bc-4933-a562-b51f18b17a75)

Reply:
I checked the electrum doc - it requires the hash of the script indeed. I thought it returns balances by the pubkey hash. I was wrong, sorry